### PR TITLE
Add simple SOS decomposition example

### PR DESCRIPTION
**Pull request by votroto**
Adds a trivial SOS decomposition example in response to issue https://github.com/jump-dev/SumOfSquares.jl/issues/166#issue-645251412
It is simple but hopefully clearer than a docstring.

### CODE REVIEW

File: examples/sos_decomposition.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using SumOfSquares
+using JuMP
+using SCS
+
+# ------------------------------------------------------------------------------
+# A trivial SOS decomposition example
+# ------------------------------------------------------------------------------
+
+# The polynomial p = x^2 - x*y^2 + y^4 + 1 is SOS.
+# Among infinite others, it has the decomposition:
```
Comment (blegat) [inline examples/sos_decomposition.jl:11]: What do you mean by "Among infinite others" ? Isn't it the only polynomial with this decomposition ?

File: examples/sos_decomposition.jl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# appropriate SOS certificate for polynomials nonnegative on semialgebraic sets.
+model = SOSModel(SCS.Optimizer)
+@constraint(model, cref, p >= 0)
+
+# Thirdly, optimize the feasibility problem! -----------------------------------
+optimize!(model)
+
+# Lastly, recover a SOS decomposition ------------------------------------------
+# In general, SOS decompositions are not unique!
+sos_decomposition = SumOfSquares.sos_decomposition(cref, 1e-4)
```
Comment (blegat) [inline examples/sos_decomposition.jl:31]: `sos_decomposition` is exported, no need for the `SumOfSquares.`

**Comment by votroto** [comment]
Does this update resolve the issues?

File: examples/sos_decomposition.jl [hunk 3]
```diff
@@ -8,8 +8,10 @@ using SCS
 # ------------------------------------------------------------------------------
 
 # The polynomial p = x^2 - x*y^2 + y^4 + 1 is SOS.
-# Among infinite others, it has the decomposition:
-# p = 3/4*(x-y^2)^2 + 1/4*(x + y)^2 + 1.
+# We can, for example, decompose it as
+# p = 3/4*(x-y^2)^2 + 1/4*(x + y)^2 + 1,
```
Comment (blegat) [inline examples/sos_decomposition.jl:12]: Add spaces around the `-` sign

File: examples/sos_decomposition.jl [hunk 4]
```diff
@@ -0,0 +1,52 @@
+using DynamicPolynomials
+using SumOfSquares
+using JuMP
+using SCS
+
```
Comment (blegat) [inline examples/sos_decomposition.jl:5]: Can you add your name here after `**Contributed by**: `? This is not done for other notebboks yet but I am reworking them at the moment

**Comment by votroto** [comment]
Sure, does username suffice as a signature?

**Comment by blegat** [comment]
> Sure, does username suffice as a signature?

Yes, as you prefer :)

**Pull request merged**